components/ca-resource.html

```diff
@@ -0,0 +1,123 @@
+<script>
+    define('ca-resource', [], () => {
+        /**
+         * Non-UI component which declares a remote resource accessed through HTTP. The value of the resource is
+         * exposed through the {@link value} property. The following events are fired by this element:
+         *
+         * <dl>
+         * <dt>ca-resource-fetching</dt><dd>Fired when the resource is about to be fetched, with the fetch options available to be modified.</dd>
+         * <dt>ca-resource-fetched</dt><dd>Fired when the resource has been fetched, with an error or non-error result.</dd>
+         * <dt>ca-resource-ready</dt><dd>Fired when the resource has been fetched successfully. {@link value} will be available.
+         * <dt>ca-resource-unauthorized</dt><dd>Fired when the resource fetch failed with 401 Unauthorised because the user's credentials could not be verified.</dd>
+         * <dt>ca-resource-forbidden</dt><dd>Fired when the resource fetch failed with 403 Forbidden because the user's credentials were not acceptable to retrieve it.</dd>
+         * </dl>
+         * TODO: make this able to PUT resources back by changing the 'value' property
+         */
+        class ApiResourceElement extends HTMLElement {
```

Comment:
The rest of the components seem to start their class names with CA. e.g. https://github.com/cambridgeweblab/common-ui/blob/master/components/ca-select.html#L79 - should this be the case?

Reply:
Yep, good spot. Copy and paste error.